pkg/controller/node/node_controller.go

```diff
@@ -963,6 +963,10 @@ func (ctrl *Controller) updateCandidateMachines(pool *mcfgv1.MachineConfigPool,
 		if err := ctrl.setDesiredMachineConfigAnnotation(node.Name, targetConfig); err != nil {
 			return goerrs.Wrapf(err, "setting desired config for node %s", node.Name)
 		}
+		ctrl.logPool(pool, "Applying %s taint to %s", constants.NodeUpdateInProgressTaint.Key, node.Name)
+		if err := ctrl.setUpdateInProgressTaint(ctx, node.Name); err != nil {
```

Comment:
Is the goal is to apply UpdateInProgress taint only to nodes that are currently being updated?
I may have misunderstood but I thought we want to apply UpdateInProgress taint to all the nodes in the pool that has not been updated yet. And scheduler will make use of this taint to ensure that we try to rescheduled drained pods to node that doesn't have that taint(implies node has already been updated).

Reply:
I thought:

All the nodes in the pool that have not been updated yet == All the nodes that doesn't have desired configuration? Is it not the case?

Reply:
Desired Configuration are not set on all node together when a new update starts. Number of nodes where we set desiredConfig is based on maxUnavailable value set. So, by default it will be set only on one node that has been picked up for update. See https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/node/node_controller.go#L770, we calculate number of candidates that can be updated before calling updateCandidateMachines()